.github/workflows/test-all-packages.yml

```diff
@@ -119,39 +119,64 @@ jobs:
       #- name: yarn test (everything)
       #  run: yarn ${{ steps.vars.outputs.test }}
       - name: yarn test (access-token)
+        if: always()
```

Comment:
these are unfortunate.. why do they become necessary?

Reply:
> these are unfortunate

There's already plenty of boilerplate in the test files, and better to be consistent than to be surgical when copy-pasta might fail to preserve the best behaviour.

> why do they become necessary?

I want all the tests to run, even if some previous one failed (but still report the job status as failure if it did).  It's been a longstanding annoyance for me that I have to push-wait-fix over and over again, gradually unveiling the errors in my code, rather than seeing all the errors my PR introduces at once the way that a local `yarn test` would have shown.

Reply:
We probably want `success() || failure()` or `! cancelled()` rather than `always()`, because the latter matches even if the job is cancelled: https://docs.github.com/en/actions/learn-github-actions/expressions#always .

Reply:
> the latter matches even if the job is cancelled
TIL that `always()` is not a synonym for `success() || failure()`. :)

Reply:
Oh, ok, I guess it makes sense that the default is to not execute step 2 if the previous step 1 failed, and in our big batch of sequential tests, we don't want that short-circuiting.

I think we'd want a few early steps (like performing the `git clone`, or rebuilding the Node cache) to be mandatory; running `cd packages/vat-data && yarn test` should only happen if the rebuild-cache step succeeded. But we don't want it to depend upon whether the previous e.g. `cd packages/swing-store && yarn test` succeeded.

Is there a way to express that? Would adding `if: success() || failure()` to each `yarn test` step accidentally cause it to attempt to run all tests even if the preliminaries failed?

Reply:
Oh damn I had forgot we had some tests that ran sequentially.

@warner I believe (but I'm not sure) that the whole `test-quick` wouldn't run if the build failed.

Reply:
> the whole `test-quick` wouldn't run if the build failed.

Correct.  All the `test-*` jobs have `needs: build` so they won't run on failure to create the build cache.  The build cache restore can fail, but that's pretty rare nowadays.